tt_metal/tt_metal.cpp

```diff
@@ -853,26 +853,21 @@ void CompileProgram(Device *device, Program &program, bool fd_bootloader_mode) {
     program.compile(device, fd_bootloader_mode);
 }
 
-void AllocateBuffer(Buffer *buffer, bool bottom_up) {
-    if(GraphTracker::instance().hook_allocate(buffer, bottom_up)) {
-        GraphTracker::instance().track_allocate(buffer, bottom_up);
+void AllocateBuffer(Buffer *buffer) {
+    if (GraphTracker::instance().hook_allocate(buffer)) {
+        GraphTracker::instance().track_allocate(buffer);
         return;
     }
-    EnqueueAllocateBuffer(buffer->device()->command_queue(), buffer, bottom_up, false);
-    GraphTracker::instance().track_allocate(buffer, bottom_up);
+    EnqueueAllocateBuffer(buffer->device()->command_queue(), buffer, false);
+    GraphTracker::instance().track_allocate(buffer);
 }
 
 void DeallocateBuffer(Buffer *buffer) {
     GraphTracker::instance().track_deallocate(buffer);
     if(GraphTracker::instance().hook_deallocate(buffer)) {
         return;
     }
-    EnqueueDeallocateBuffer(
-        buffer->device()->command_queue(),
-        *(buffer->device()->allocator_),
-        buffer->address(),
-        buffer->buffer_type(),
-        false);
+    EnqueueDeallocateBuffer(buffer->device()->command_queue(), *(buffer->device()->allocator_), buffer, false);
```

Comment:
I wonder if it makes sense to add `Buffer:get_allocator()` so same pattern `*(buffer->device()->allocator_)` which we use in many places could be simplified.

Reply:
Hmm, normally most code doesn't directly interact with allocator, there should only be two places, the initial allocate and deallocate buffer calls. Seems like we're calling it in some places where we already have a reference to the allocator, I will clean those up so there's only 2 for now and if it's more common we can add an api